kyo-core/shared/src/main/scala/kyo/layers.scala

```diff
@@ -0,0 +1,144 @@
+package kyo
+
+object layers {
+
+  trait Layer[Sin, Sout] { self =>
```

Comment:
It was a bit counterintuitive to me that the effects that the layer can handle is "out" and effects it can produce is "in". I image that's thinking of the operations in the pending effect set. What do you think about flipping the meaning of the type parameters? I think it's more intuitive to think of a `Layer` like an effect-level arrow that takes effects to handle and produces new effects if needed.

Reply:
Yeah I agree. How about `Layer[Handled, S]` or something along those lines, where `Handled` are the effects that are handled and `S` are the new unhandled effects?

Reply:
Maybe `Input`/`Output` to make it more explicit?

Reply:
It's now `Layer[In, Out]`, where `In` is the intersection of all effects the layer will handle, and `Out` is the intersection of the effects that will remain to be handled.